src/commands.rs

```diff
@@ -4935,14 +4935,31 @@ fn cmd_git_push(
             old_heads.extend(old_head.adds());
         }
     }
+    if old_heads.is_empty() {
+        old_heads.push(repo.store().root_commit_id().clone());
+    }
     for index_entry in repo.index().walk_revs(&new_heads, &old_heads) {
         let commit = repo.store().get_commit(&index_entry.commit_id())?;
-        if commit.tree().has_conflict() {
-            return Err(UserError(format!(
-                "Won't push commit {} since it has conflicts",
-                short_commit_hash(commit.id())
-            )));
-        }
+        let reason = if commit.description().is_empty() {
+            "it has no description"
+        } else if commit.author().name == jujutsu_lib::settings::USER_NAME_PLACEHOLDER
+            || commit.author().email == jujutsu_lib::settings::USER_EMAIL_PLACEHOLDER
+        {
+            "it has no author set"
+        } else if commit.committer().name == jujutsu_lib::settings::USER_NAME_PLACEHOLDER
+            || commit.committer().email == jujutsu_lib::settings::USER_EMAIL_PLACEHOLDER
+        {
+            "it has no committer set"
+        } else if commit.tree().has_conflict() {
+            "it has conflicts"
+        } else {
+            continue;
```

Comment:
I'm not a fan of putting flow control inside what is essentially an assignment statement.  I'd prefer if reason were an empty string here and the flow control was handled separately.

Reply:
I actually had done it that way before but with `Option` so there's no risk of using the sentinel value. Then I switched to this form to avoid the repeated `Some()`. I'll switch back.